crates/block-producer/src/block_producer.rs

```diff
@@ -483,17 +576,20 @@ impl BlockProducer {
             .extend(generated_withdrawal_cells.outputs);
 
         // reverted withdrawal cells
-        if let Some(reverted_withdrawals) = crate::withdrawal::revert(
+        if let Some(mut reverted_withdrawals) = crate::withdrawal::revert(
             &rollup_action,
             rollup_context,
             &self.config,
             &self.rpc_client,
         )
         .await?
         {
-            tx_skeleton
-                .cell_deps_mut()
-                .extend(reverted_withdrawals.deps);
+            let mut resolved_deps =
+                resolve_type_deps(&self.rpc_client, &reverted_withdrawals.inputs).await?;
+            resolved_deps.extend(tx_skeleton.cell_deps_mut().drain(..));
+            resolved_deps.extend(reverted_withdrawals.deps.drain(..));
+
+            *tx_skeleton.cell_deps_mut() = resolved_deps.into_iter().collect();
```

Comment:
Why not just call `tx_skeleton.cell_deps_mut().extend(...)`?

Reply:
Done